src/main.c

```diff
@@ -1882,13 +1904,18 @@ static int listener_setup_ssl(h2o_configurator_command_t *cmd, h2o_configurator_
                                            "identity:a,certificate-file:s,key-file:s,min-version:s,minimum-version:s,max-version:s,"
                                            "maximum-version:s,cipher-suite:s,cipher-suite-tls1.3:a,ocsp-update-cmd:s,"
                                            "ocsp-update-interval:*,ocsp-max-failures:*,dh-file:s,cipher-preference:*,neverbleed:*,"
-                                           "http2-origin-frame:*,client-ca-file:s,ech:a",
+                                           "http2-origin-frame:*,client-ca-file:s,ech:a,verify-peer:s",
                                            &identity_node, &certificate_file, &key_file, &min_version, &min_version, &max_version,
                                            &max_version, &cipher_suite, &cipher_suite_tls13_node, &ocsp_update_cmd,
                                            &ocsp_update_interval_node, &ocsp_max_failures_node, &dh_file, &cipher_preference_node,
-                                           &neverbleed_node, &http2_origin_frame_node, &client_ca_file, &ech_node) != 0)
+                                           &neverbleed_node, &http2_origin_frame_node, &client_ca_file, &ech_node, &verify_peer_node) != 0)
             return -1;
         if (identity_node != NULL) {
+            if (is_reverse_listener(listener)) {
+                h2o_configurator_errprintf(cmd, *identity_node,
+                                           "identity directive is not supported in reverse listener");
+                return -1;
+            }
```

Comment:
I think we can negotiate the type of client certificates, but we do not need to do that at this stage.